### ISSUE TITLE
Structs with fields starting with underscores are not readable and clash with non-underscore fields

### ISSUE THREAD

**Issue by antazoey** [ethereum/web3.py#3429]
### What happened?

I have a Struct in solidity with two fields, `a` and `_a`. When I try to use `decode_tuples=True` and call the method returning it, the `_a` value replaces `a` and I cannot refer to `_a` anywhere.

See the script below for a simplified, reproducible example!

### Code that produced the error

```python
from web3 import Web3
from solcx import compile_source

code = """
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.2;

contract SolidityContract {
    struct MyStruct {
        address _a;
        uint256 a;
    }

    function getStruct() public view returns(MyStruct memory) {
        return MyStruct(msg.sender, 123);
    }
}
"""

def main():
    print("Compiling...")
    compiled_sol = compile_source(code)
    contract_id, contract_interface = compiled_sol.popitem()
    bytecode = contract_interface['bin']
    abi = contract_interface['abi']
    w3 = Web3(Web3.EthereumTesterProvider())
    w3.eth.default_account = w3.eth.accounts[0]
    contract = w3.eth.contract(abi=abi, bytecode=bytecode)

    print("Deploying...")
    tx_hash = contract.constructor().transact()
    tx_receipt = w3.eth.wait_for_transaction_receipt(tx_hash)
    instance = w3.eth.contract(
        address=tx_receipt.contractAddress,
        abi=abi,
        decode_tuples=True,  # Ensure we are using the functionality!
    )

    print("Calling method with struct with underscore in name...")
    result = instance.functions.getStruct().call()
    try:
        print(f"SUCCESS: {result.a}")
    except AttributeError as err:
        print(f"Can't refer to 'a': {err}")

    try:
        print(f"SUCCESS: {result._a}")
    except AttributeError as err:
        print(f"Can't refer to '_a': {err}")


if __name__ == "__main__":
    main()
```

### Full error output

```shell
Compiling...
Deploying...
Calling method with struct with underscore in name...
SUCCESS: 123
Can't refer to '_a': 'ABIDecodedNamedTuple' object has no attribute '_a'

# NOTES:
1. 123 is the value for `_a` and not `a` but it shows up in `a`.
2. Referring to `_a` is not possible.
```

### Fill this section in if you know how this could or should be fixed

I am not sure! Namedtuples don't allow underscore prefixed values so it may not be the best data-structure to use (without additional hacks)

### web3 Version

7.0.0b7

### Python Version

3.12.2

### Operating System

macOS

### Output from `pip freeze`

```shell
tldr
```

**Comment by pacrob** [comment]
Testing locally, it looks like it's working as expected. You have `address _a` and `uint256 a`. `a` comes out as `123` as expected, and `_a` is renamed to `_0` as per the namedtuple [docs](https://docs.python.org/3/library/collections.html#collections.namedtuple)
